crates/sui-indexer/src/db.rs

```diff
@@ -12,16 +13,27 @@ use diesel::PgConnection;
 pub type ConnectionPool = Pool<ConnectionManager<PgConnection>>;
 pub type PoolConnection = PooledConnection<ConnectionManager<PgConnection>>;
 
-#[derive(Debug, Clone, Copy)]
+#[derive(Args, Debug, Clone)]
 pub struct ConnectionPoolConfig {
+    #[arg(long, default_value_t = 100)]
+    #[arg(env = "DB_POOL_SIZE")]
     pub pool_size: u32,
+    #[arg(long, value_parser = parse_duration, default_value = "30")]
+    #[arg(env = "DB_CONNECTION_TIMEOUT")]
     pub connection_timeout: Duration,
+    #[arg(long, value_parser = parse_duration, default_value = "3600")]
+    #[arg(env = "DB_STATEMENT_TIMEOUT")]
     pub statement_timeout: Duration,
 }
 
+fn parse_duration(arg: &str) -> Result<std::time::Duration, std::num::ParseIntError> {
+    let seconds = arg.parse()?;
+    Ok(std::time::Duration::from_secs(seconds))
+}
+
 impl ConnectionPoolConfig {
     const DEFAULT_POOL_SIZE: u32 = 100;
-    const DEFAULT_CONNECTION_TIMEOUT: u64 = 3600;
+    const DEFAULT_CONNECTION_TIMEOUT: u64 = 30;
```

Comment:
Curious for the context on this change -- I vaguely recall there was some reason why @gegaowp set this to a large timeout previously.

Reply:
So there's a difference between Connection Timeout (the time we wait to establish a new connection) and Statement Timeout which is the time we allow for a single SQL statement to take. Having an hour long timeout for establishing a connection seems...not great?